p2p/protocol/identify/obsaddr.go

```diff
@@ -382,16 +382,19 @@ func (oas *ObservedAddrManager) recordObservationUnlocked(conn network.Conn, obs
 	observerString := observerGroup(conn.RemoteMultiaddr())
 	localString := string(conn.LocalMultiaddr().Bytes())
 	ob := observation{
-		seenTime:      now,
-		connDirection: conn.Stat().Direction,
+		seenTime: now,
+		inbound:  conn.Stat().Direction == network.DirInbound,
 	}
 
-	observedAddrs := oas.addrs[localString]
 	// check if observed address seen yet, if so, update it
-	for i, previousObserved := range observedAddrs {
-		if previousObserved.addr.Equal(observed) {
-			observedAddrs[i].seenBy[observerString] = ob
-			observedAddrs[i].lastSeen = now
+	for _, observedAddr := range oas.addrs[localString] {
+		if observedAddr.addr.Equal(observed) {
+			// Don't trump an outbound observation with an inbound
+			// one.
```

Comment:
Do you mean the opposite? You're saying to mark as inbound if it was, or is currently inbound.

Reply:
Fixed, I think.

Reply:
I think the wording here is just throwing me off (maybe just tired 😃). Maybe something like "Observation is considered inbound if the peer ever gave it to us as an inbound observation"